ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_bgp.py

```diff
@@ -399,8 +399,41 @@ def _router_bgp_vlan_aware_bundles(self) -> list | None:
                     )
 
             # L2 Vlans per Tenant
+            # If multiple L2 Vlans share the same evpn_vlan_bundle name, they will be part of the same vlan-aware-bundle
+            l2vlans = list(tenant["l2vlans"])
+            l2vlans_no_bundle_definition = list()
+            if len(l2vlans) > 0:
+                tmp_bundles = {}
+                for l2vlan in l2vlans:
+                    if "evpn_vlan_bundle" in l2vlan:
+                        # vlans which have evpn_vlan_bundle defined
+                        if l2vlan["evpn_vlan_bundle"] not in tmp_bundles:
+                            tmp_bundles[l2vlan["evpn_vlan_bundle"]] = {"vlans":[], "bundle_details":{}}
+                        
+                        if (len(tmp_bundles[l2vlan["evpn_vlan_bundle"]]["bundle_details"]) < 1) and ((self._router_bgp_vlans_vlan(l2vlan, tenant, vrf={})) is not None):
+                            # We are reusing the regular bgp vlan function so need to add vlan info  
+                            tmp_bundles[l2vlan["evpn_vlan_bundle"]]["bundle_details"] = self._router_bgp_vlans_vlan(l2vlan, tenant, vrf={})
+                            
+                        tmp_bundles[l2vlan["evpn_vlan_bundle"]]["vlans"].append(int(l2vlan["id"]))
+                    else:
+                        # populate list for l2vlans which don't have evpn_vlan_bundle defined, these will be bundled based on vlan name in a later section
+                        l2vlans_no_bundle_definition.append(l2vlan)
+                            
+                for bundle_name, values in tmp_bundles.items():
+                    bundle = {}       
+                    bundle["vlan"] = list_compress(values["vlans"])
+                    bundle["name"] = str(bundle_name)
+                    bundle.update(values["bundle_details"])
+                    append_if_not_duplicate(
+                        list_of_dicts=bundles,
+                        primary_key="name",
+                        new_dict=bundle,
+                        context="BGP VLAN-Aware Bundles defined under network services",
+                        context_keys=["name"],
+                    )            
+            
```

Comment:
Instead of adding all this extra logic which will be a bit "expensive", I think we should just create a small function we can give to the `groupby` below instead of the static key `"name"`. I believe `groupby` supports a callable as `key` so you can write the function to get `evpn_vlan_bundle` key if present, or fallback to the `name` key. This way we get less code duplication, and it will run much more efficiently.